### PR TITLE
TandemFoil Paper: gc=0.7 at champion config (tighter clip)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Tighter clipping (gc=0.7 vs 0.5 champion) may delay or prevent the ep462 divergence pattern by more aggressively constraining gradient updates at cosine LR peaks.

## Instructions

```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil_paper --optimizer lion --lr 1.25e-4 --cosine-t-max 10 \
  --grad-clip 0.7 --weight-decay 1e-2 --enable-fourier \
  --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 --ema-decay 0.999 \
  --wandb_group tfp-gc-sweep
```

## Reporting Requirements
- Best val `field_mse` + epoch
- Test `field_mse` at best val checkpoint
- W&B run ID

## Baseline

**TFP baselines:**
- val best: `field_mse = 0.002383` @ ep443 (PR #3025, W&B: `d1xh0o1p`)
- Config: Lion lr=1.25e-4, T_max=10, gc=0.5, EMA=0.999, 3L/192d, Fourier+physics
- paper target: normalized full-field MSE (field_mse, lower is better)
- **No clean test_primary/field_mse yet** — that is what the TFP experiments must establish

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT nobara:

## Results

### gc=0.7 on TandemFoil Paper — NEGATIVE RESULT

| Metric | gc=0.7 (this run) | gc=0.5 baseline (#3025) |
|--------|-------------------|------------------------|
| Best val `field_mse` | **Infinity** (never converged) | **0.002383** @ ep443 |
| Divergence epoch | **ep142** (catastrophic) | ep462 |
| Pre-divergence loss | 0.047 @ ep110 | — |
| Pre-divergence surface Ux | 0.0013 @ ep107 | — |

**W&B run:** `e0am2f3z` (nobara/tfp-gc-07, group: tfp-gc-sweep)

**Command:**
```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil_paper --optimizer lion --lr 1.25e-4 --cosine-t-max 10 \
  --grad-clip 0.7 --weight-decay 1e-2 --enable-fourier \
  --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 --ema-decay 0.999 \
  --agent nobara --wandb-name "nobara/tfp-gc-07" \
  --wandb-group tfp-gc-sweep
```

**Peak memory:** ~36 GB (GPU 0)

### What happened

gc=0.7 **catastrophically diverged at epoch 142**, far earlier than the gc=0.5 baseline which was stable until ep462. The `field_mse` metric **never became finite** — pressure predictions (surface_mse_p) were still oscillating between `e+16` and `e+32` when divergence struck.

**Timeline:**
- **Epochs 1-113:** Healthy training. Loss descended from 0.946 → 0.047. Non-pressure metrics (Ux, Uy) improved steadily. Pressure predictions slowly converging (surface_mse_p trough envelope: `e+31` → `e+16`), but still far from meaningful values.
- **Epoch 114:** First instability spike (loss 0.080 → 0.122). Model partially recovered by ep120.
- **Epochs 120-140:** Repeated instability cycles every ~10 epochs (ep122, ep126, ep132, ep141). Each spike disrupted forward progress, with the model spending half its time recovering.
- **Epoch 142-144:** Terminal divergence. Loss exploded 0.147 → 0.417 → 0.589, grad norms 27 → 78 → 256. Model reverted past epoch-1 performance.

**Key insight:** gc=0.7 is **too loose** for TFP with Lion + T_max=10. The gradient clip value directly controls how much damage cosine LR peaks can do. gc=0.5 constrains gradients enough to survive 462 epochs of T_max=10 cycling; gc=0.7 allows enough extra gradient magnitude to trigger cascading instability by ep114. The 3x earlier divergence (ep142 vs ep462) is a dramatic stability gap for just a 0.2 change in clip value.

**Hypothesis verdict:** FALSIFIED. gc=0.7 is strictly worse than gc=0.5 — not only did it diverge 3x earlier, it never even reached finite field_mse. gc=0.5 remains the optimal clip value.

### Suggested follow-ups
- **gc=0.3 sweep** — if gc=0.5 > gc=0.7, even tighter clipping might extend stability further and push field_mse below 0.002383
- **gc=0.5 + T_max=20** — longer cosine periods reduce peak LR frequency, which could extend the stable training window beyond ep462

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3091 because ADVISOR: Closing — gc=0.7 is strictly worse than gc=0.5 on TFP. val_primary/field_mse was Infinity for all 177 epochs. Diverged 3x earlier than champion (ep142 vs ep462) with grad norms reaching 1,326. Relaxing gradient clipping beyond 0.5 destabilizes Lion+T_max=10. The gc=0.3 experiment (#3090 nezuko) tests the tighter direction, which is the more promising path. Good clean result, nobara..
